locator/token_metadata.cc

```diff
@@ -522,39 +513,40 @@ void token_metadata_impl::debug_show() const {
 
 void token_metadata_impl::update_host_id(const host_id& host_id, inet_address endpoint) {
     _topology.add_or_update_endpoint(endpoint, host_id);
-    _endpoint_to_host_id_map[endpoint] = host_id;
 }
 
 host_id token_metadata_impl::get_host_id(inet_address endpoint) const {
-    if (!_endpoint_to_host_id_map.contains(endpoint)) {
+    if (const auto* node = _topology.find_node(endpoint)) [[likely]] {
+        return node->host_id();
+    } else {
         throw std::runtime_error(format("host_id for endpoint {} is not found", endpoint));
     }
-    return _endpoint_to_host_id_map.at(endpoint);
 }
```

Comment:
I think we should jut remove these helper methods and add an accessor to `topology` and have users use it to find nodes directly. Can be a follow-up.

Reply:
token_metadata::get_topology() already exists so the callers can be converted to call get_topology().find_node() on their own.
Sounds like a good follow up.